lib/slack-ruby-bot/commands/base.rb

```diff
@@ -1,7 +1,69 @@
 module SlackRubyBot
   module Commands
+
+    # class methods to modify the Commands::Base class. All methods are related to
+    # providing before/after hooks to commands.
+    module CommandHooks
+      def remove_command_hooks(type = nil, name = nil)
```

Comment:
Tests do rely on it but this is required functionality. We need to be able to remove hooks/callbacks during runtime. Note that `ActiveSupport::Callbacks` also provides a facility to remove callbacks. I do not think this is superfluous.

However, I am open to a rename of the method. `reset!` doesn't fit since the method takes args to remove by type (`:before` & `:after`) and by 'verb'. Maybe `clear`?